specs/CookieManagement.md

```diff
@@ -0,0 +1,505 @@
+  /// Note that, for instance, ".bing.com", "bing.com", and "www.bing.com" are
+  /// considered different domains.
+  [propget] HRESULT Domain([out, retval] LPWSTR* domain);
+
+  /// The path for which the cookie is valid. The default is "/", which means
+  /// this cookie will be sent to all pages on the Domain.
+  [propget] HRESULT Path([out, retval] LPWSTR* path);
+
+  /// The expiration date and time for the cookie as the number of seconds since the UNIX epoch.
+  /// The default is -1.0, which means cookies are session cookies by default.
+  [propget] HRESULT Expires([out, retval] double* expires);
+  /// Set the Expires property. Cookies are session cookies
+  /// and will not be persistent if Expires is negative.
+  [propput] HRESULT Expires([in] double expires);
+
+  /// Whether this cookie is http-only.
+  /// True if a page script or other active content cannot access this
+  /// cookie. The default is false.
+  [propget] HRESULT IsHttpOnly([out, retval] BOOL* isHttpOnly);
+  /// Set the IsHttpOnly property.
+  [propput] HRESULT IsHttpOnly([in] BOOL isHttpOnly);
+
+  /// SameSite status of the cookie which represents the enforcement mode of the cookie.
+  /// The default is COREWEBVIEW2_COOKIE_SAME_SITE_KIND_LAX.
+  [propget] HRESULT SameSite([out, retval] COREWEBVIEW2_COOKIE_SAME_SITE_KIND* sameSite);
+  /// Set the SameSite property.
+  [propput] HRESULT SameSite([in] COREWEBVIEW2_COOKIE_SAME_SITE_KIND sameSite);
+
+  /// The security level of this cookie. True if the client is only to return
+  /// the cookie in subsequent requests if those requests use HTTPS.
+  /// The default is false.
+  /// Note that cookie that requests COREWEBVIEW2_COOKIE_SAME_SITE_KIND_NONE but
+  /// is not marked Secure will be rejected.
+  [propget] HRESULT IsSecure([out, retval] BOOL* isSecure);
+  /// Set the IsSecure property.
+  [propput] HRESULT IsSecure([in] BOOL isSecure);
+
+  /// Whether this is a session cookie. The default is false.
+  [propget] HRESULT IsSession([out, retval] BOOL* isSession);
+}
+
+/// Creates, adds or updates, gets, or or view the cookies. The changes would
+/// apply to the context of the user profile. That is, other WebViews under the
+/// same user profile could be affected.
+[uuid(588C8A15-A28A-4FFD-926B-5E6EE7449E7C), object, pointer_default(unique)]
+interface ICoreWebView2CookieManager : IUnknown {
+  /// Create a cookie object with a specified name, value, domain, and path.
+  /// One can set other optional properties after cookie creation.
+  /// This only creates a cookie object and it is not added to the cookie
+  /// manager until you call AddOrUpdateCookie.
+  /// See ICoreWebView2Cookie for more details.
+  HRESULT CreateCookieWithDetails(
+    [in] LPCWSTR name,
+    [in] LPCWSTR value,
+    [in] LPCWSTR domain,
+    [in] LPCWSTR path,
+    [out, retval] ICoreWebView2Cookie** cookie);
+
+  /// Creates a cookie whose params matches those of the specified cookie.
+  HRESULT CreateCookie(

```

Comment:
Should this be called CopyCookie?

Reply:
This should remain CreateCookie while the other with more params should be called CreateCookieWithDetails.